_drafts/gdbundle.md

```diff
@@ -0,0 +1,526 @@
+many mediocre vendor SDK's. Tens of thousands of developers use each one of
+these stacks/libraries, and every single one of them has to manually debug each
+module by hand by using GDB's print functionality or write their own scripts.
+
+This is why embedded developers often choose to use thousand dollar or mediocre
+IDE debuggers over using GDB. They have these debugging utilities built in or
+allow extensions to be integrated and sold[^code_confidence], even though the
+software backing them isn't all that complex.
+
+### Poor Installation and Non-Discoverable of Extensions
+
+In the age of modern developer tools, easy installation of extensions is a
+critical facet that needs to be taken seriously. Let's take the new wave of text
+editors as an example. The ones that have gained massive popularity over the
+last 5 years, Sublime Text, VSCode, and Atom, all have a built-in package
+manager which allows searching for and installing extensions with only a few
+keystrokes. **This is what GDB should have**.
+
+As for discoverability, I love websites like
+[VimAwesome](https://vimawesome.com/) and the
+[VSCode Marketplace](https://marketplace.visualstudio.com/vscode). I check them
+regularly to see which pieces of my workflows have been automated for me.
+
+If you believe this ease of extensibility is only for high-level applications
+and desktop style software, check out
+[PlatformIO's Libraries page](https://platformio.org/lib/search?query=) which
+provides easy installation and discoverability of compiled C/C++ packages for
+embedded systems and Arduino.
+
+### Monolithic, Invasive Projects
+
+Since there is no trivial way for GDB scripts to build upon each other, all
+ambitious projects, such as [hugsy/gef](https://github.com/hugsy/gef),
+[pwndbg/pwndbg](https://github.com/pwndbg/pwndbg), and
+[snare/voltron](https://github.com/snare/voltron), are required to rewrite or
+copy-paste functionality that would normally be imported from popular Python
+packages (e.g. termcolor). This is a barrier to entry for plugin developers.
+
+On the other side of the pendulum, since packages can't be imported easily
+through the Python ecosystem, users are required to run
+[invasive](https://github.com/pwndbg/pwndbg/blob/dev/setup.sh)
+[shell](https://github.com/snare/voltron/blob/master/install.sh) scripts that
+use the system's package manager and install Python packages into the system's
+Python installation.
+
+Every developer should be using virtual environments and leaving their system
+Python installation alone, and the Python environment within GDB is no
+different.
+
+### Lack of Customization
+
+On one hand, some might see this as a nice feature of GDB. Once you get a GDB
+shell, you know exactly how to accomplish what you want, assuming you know how
+to use GDB to begin with. This is similar to how people think about Vi(m).
+
+However, Vim is infinitely better today due to its customization options, plugin
```

Comment:
Again, you should more explicitly call out the value: GDB is a pro tool with many different use cases. Plugins make it much more flexible and allow it to better fit those use cases.

Then you can use Vim a an example of that (e.g. plugins for support of smaller / less popular languages).